cs251tk/toolkit/__main__.py

```diff
@@ -102,7 +102,14 @@ def main():
         sys.exit(1)
 
     for spec_to_use in assignments:
-        check_dependencies(specs[spec_to_use])
+        try:
+            check_dependencies(specs[spec_to_use])
+        except KeyError:
+            print('Spec {} does not exist'.format(spec_to_use), file=sys.stderr)
+            assignments.remove(spec_to_use)
```

Comment:
Mutating data structures while iterating throws an error in my testing:

```
>>> foo=set([1,2,3])
>>> foo
{1, 2, 3}
>>> for x in foo:
...   foo.remove(x)
...
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: Set changed size during iteration
```

Reply:
Maybe something like this?

```python3
for ...:
  except KeyError
    print('Spec {} does not exist'.format(spec_to_use), file=sys.stderr)
    missing.add(spec_to_use)

assignments = assignments.difference(missing)

if not assignments:
  print('no valid specs remaining', file=sys.stderr)
  sys.exit(1)
```

Reply:
What about this?
```
    available_specs = assignments

    for spec_to_use in assignments:
        try:
            check_dependencies(specs[spec_to_use])
        except KeyError:
            print('Spec {} does not exist'.format(spec_to_use), file=sys.stderr)
            available_specs.remove(spec_to_use)

    assignments = available_specs

    if not assignments:
        print('no valid specs remaining', file=sys.stderr)
        sys.exit(1)
```

Reply:
SGTM